apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -228,21 +231,32 @@ export class SubpoenaService {
       )
     }
 
-    return this.findBySubpoenaId(subpoena.subpoenaId)
+    return this.findById(subpoena.id)
   }
 
-  async findBySubpoenaId(subpoenaId?: string): Promise<Subpoena> {
-    if (!subpoenaId) {
-      throw new Error('Missing subpoena id')
+  async findById(subpoenaId: string): Promise<Subpoena> {
+    const subpoena = await this.subpoenaModel.findOne({
+      include,
+      where: { id: subpoenaId },
+    })
+
+    if (!subpoena) {
+      throw new NotFoundException(`Subpoena ${subpoenaId} does not exist`)
     }
 
+    return subpoena
+  }
+
+  async findBySubpoenaId(policeSubpoenaId?: string): Promise<Subpoena> {
     const subpoena = await this.subpoenaModel.findOne({
       include,
-      where: { subpoenaId },
+      where: { subpoenaId: policeSubpoenaId },
```

Comment:
In hindsight maybe this should've just been policeSubpoenaId in the database as well to avoid all confusion, we could maybe change it at some point but not in this PR.

Reply:
Agree, a good cooldown task